airflow/api_fastapi/parameters.py

```diff
@@ -171,7 +180,10 @@ def to_orm(self, select: Select) -> Select:
                 f"the attribute does not exist on the model",
             )
 
-        column = self.attr_mapping.get(lstriped_orderby, None) or getattr(DagModel, lstriped_orderby)
+        if self.to_replace:
+            lstriped_orderby = self.to_replace.get(lstriped_orderby, lstriped_orderby)
```

Comment:
I think we can get rid of the `to_replace`. It adds extra complexity for mapping things to gether. We can just rename to use the appropriate one. No need to remap. i.e `conn_id`. Or again find a way to always just use the mapping. So we have only 1 ugly mapping and everything uses that to retrieve the `db/model` equivalent.

Reply:
I realized the legacy endpoint returns column value as `connection_id` in the return. Since it should be backwards compatible, I have adjusted it accordingly. I think `to_replace` was a bit related to that part. We should change the `to_replace` cases as in the models below. 

`Field(alias="conn_id")` -> `Field(serialization_alias="connection_id", validation_alias="conn_id")`